### PR TITLE
Allow customizing initial date in scenarios

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: Features "Allow customizing initial date in scenarios"```

#### Purpose of change

Allow customizing initial date in scenarios.

#### Describe the solution

Get rid of scenario season flag in favor of more granular initial dates customization.

I've also updated scenario selection UI code and updated documentation a bit.

#### Describe alternatives you've considered

None.

#### Testing

1. Create a custom scenario to start on desired hour, day, month and/or year.

#### Additional context

**Usage example:**

```JSON
"custom_initial_date": { "hour": 3, "day": 10, "season": "winter", "year": 1 }
```

**Screenshots:**

image

image

### CODE REVIEW

File: src/newcharacter.cpp [hunk 1]
```diff
@@ -2260,26 +2282,46 @@ tab_direction set_scenario( avatar &u, points_left &points,
...
+                if( !sorted_scens[cur_id]->random_initial_season() ) {
+                    wprintz( w_initial_date, c_light_gray, _( "Season: %s" ),
+                             calendar::name_season( sorted_scens[cur_id]->initial_season() ) );
+                    wprintz( w_initial_date, c_light_gray, ( "\n" ) );
+                } else {
+                    wprintz( w_initial_date, c_light_gray, _( "Season: Random" ) );
+                    wprintz( w_initial_date, c_light_gray, ( "\n" ) );
+                }
+                if( !sorted_scens[cur_id]->random_initial_day() ) {
+                    wprintz( w_initial_date, c_light_gray, _( "Day: %d" ), sorted_scens[cur_id]->initial_day() );
```
Comment (ZhilkinSerg) [inline src/newcharacter.cpp:2304]:
```suggestion
                    wprintz( w_initial_date, c_light_gray, _( "Day:    %d" ), sorted_scens[cur_id]->initial_day() );
```

**Comment by Fris0uman** [comment]
Can you add a doc entry for `"custom_initial_date":` ?

File: doc/JSON_INFO.md [hunk 2]
```diff
@@ -3732,6 +3732,22 @@ Add a map special to the starting location, see JSON_FLAGS for the possible spec
 
 A list of mission ids that will be started and assigned to the player at the start of the game. Only missions with the ORIGIN_GAME_START origin are allowed. The last mission in the list will be the active mission, if multiple missions are assigned.
 
+## `custom_initial_date`
+(optional, object with optional members "hour", "day", "season" and "year")
+
+Allows customizing initial date. If not set - corresponding values from world options are used. Random value is used for each parameter that is not explicitly.
```
Comment (Fris0uman) [inline doc/JSON_INFO.md:3738]:
```suggestion
Allows customizing initial date. If not set - corresponding values from world options are used. Random value is used for each parameter that is not explicitly set in world options.
```
I think there's something missing at the end of this sentence.  Is this suggestion correct?

File: doc/JSON_INFO.md [hunk 3]
```diff
@@ -3677,127 +3693,51 @@ The following properties (mandatory, except if noted otherwise) are supported:
...
+
+* `TYPE` - The provided string must completely match the base type id of the
+  overmap terrain id, which means that suffixes for rotation and linear terrain
+  types are ignored.
+
+* `PREFIX` - The provided string must be a complete prefix (with additional
+  parts delimited by an underscore) of the overmap terrain id. For example,
+  "forest" will match "forest" or "forest_thick" but not "forestcabin".
 
 Arbitrary flags. TODO: document them.
```
Comment (kevingranade) [inline doc/JSON_INFO.md:3723]: Bad merge fixup.
```suggestion
```

File: doc/JSON_INFO.md (hunk 2 above)
Comment (kevingranade) [reply doc/JSON_INFO.md:3661]: He means, "If there is a custom_initial_date entry, each sub-memeber that is not set will get a random value".

File: src/newcharacter.cpp [hunk 4]
```diff
@@ -2094,18 +2094,39 @@ tab_direction set_scenario( avatar &u, points_left &points,
...
-        w_profession = catacurses::newwin( 4, ( TERMX / 2 ) - 1, point( TERMX / 2, 7 ) );
-        w_location = catacurses::newwin( 3, ( TERMX / 2 ) - 1, point( TERMX / 2, 11 ) );
-        w_vehicle = catacurses::newwin( 3, ( TERMX / 2 ) - 1, point( TERMX / 2, 14 ) );
-        // 11 = 2 + 4 + 3 + 3, so we use rest of space for flags
-        w_flags = catacurses::newwin( iContentHeight - 14, ( TERMX / 2 ) - 1,
-                                      point( TERMX / 2, 17 ) );
+        const int second_column_w =  TERMX / 2 - 1;
+        const int second_column_origin_x = second_column_w + 1;
+        const int second_column_origin_y = 5;
+        point origin = point( second_column_origin_x, second_column_origin_y );
```
Comment (kevingranade) [inline src/newcharacter.cpp:2106]:
```suggestion
        point origin = point( second_column_w + 1, 5 );
```

**Pull request merged**